### PR TITLE
 Context manager support with connection object

### PR DESCRIPTION
**Pull request by Earammak**
Is your feature request related to a problem? Please describe.
When working with IO, It is common to acquire some resource i.e open('file.txt') or ibm_db_dbi.connect(connection_str). These resources also have to manually closed i.e fp.close() or connection.close(). Developers may forget to clean up this resources and this could lead to resource leaks. Python solves this problem with context managers which automatically solve the problem of resource management and ensure proper cleanup. They provide a convenient and reliable way to allocate and release resources, such as connecting and disconnecting from a database.

Context managers could provide a concise and intuitive way to handle database connections and transactions, ensuring proper cleanup and reducing the risk of resource leaks. With context managers, developers can write cleaner and more readable code by encapsulating the database-related operations within a well-defined scope. Additionally, context managers can also enable automatic rollback of transactions in case of exceptions, ensuring data consistency and integrity.

Describe the solution you'd like
Context managers to be added to connection objects such that they can be used easily like:

with ibm_db_dbi.connect(connection_str) as connection:
cursor = connection.cursor()
cursor.execute('SELECT * from table')
result = cursor.fetchall()
This removes the need for manual cleanup of resources

Describe alternatives you've considered

Additional context

### CODE REVIEW

File: ibm_db_ctx.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        """Instantiate a DB2 connection."""
+        #print("init method called")
+        self.__dsn = dsn
+        self.__username = username
+        self.__password = password
+        self.__cxn = None
+
+    def __enter__(self) -> 'IBM_DBConnection':
+        """Connect to DB2."""
+        self.__cxn = ibm_db.connect(self.__dsn, '', '')
```
Comment (jaeger-2601) [inline ibm_db_ctx.py:17]: Please put username and password here

Comment (Earammak) [reply ibm_db_ctx.py:17]: > Please put username and password here

Username and password will be there in DSN only. 
For cataloged connection, we need to give database, username and password [ For this new test suite added for both cataloged and uncatalogued connection].

Comment (jaeger-2601) [reply ibm_db_ctx.py:17]: Ok. Then why are we taking username and password in the constructor then. I don't see those values being use anywhere?

Comment (Earammak) [reply ibm_db_ctx.py:17]: The use of username and password depends on what type of connection we are trying to. 
When we go for an uncatalogued connection, we will pass an empty string only for username and password field.
Please refer to:  https://github.com/ibmdb/python-ibmdb/wiki/APIs#ibm_dbconnect

Comment (jaeger-2601) [reply ibm_db_ctx.py:17]: ok

**Review by jaeger-2601 — APPROVED** [review]

**Review by bimalkjha — APPROVED** [review]

**Pull request merged**